### DISCUSSION TITLE
Unable to switch windows (kitty, lazyvim)

### DISCUSSION THREAD

**Discussion by klnstprx** [mrjones2014/smart-splits.nvim#236] [Q&A] [answered]
I am trying to get smart-splits to work with kitty. 
I have installed smart-splits using a snippet recommened inside the [LazyVim for Ambitious Developers](https://lazyvim-ambitious-devs.phillips.codes/) book.
```
return {
  "mrjones2014/smart-splits.nvim",
  build = "./kitty/install-kittens.bash",
  keys = {
    {
      "<C-h>",
      function()
        require("smart-splits").move_cursor_left()
      end,
      desc = "Move to left window",
    },
    {
      "<C-l>",
      function()
        require("smart-splits").move_cursor_right()
      end,
      desc = "Move to right window",
    },
    {
      "<C-j>",
      function()
        require("smart-splits").move_cursor_down()
      end,
      desc = "Move to below window",
    },
    {
      "<C-k>",
      function()
        require("smart-splits").move_cursor_up()
      end,
      desc = "Move to above window",
    },
  },
}
```

Then I added the following to my kitty conf file:
```
# smart splits integration
map ctrl+j neighboring_window down
map ctrl+k neighboring_window up
map ctrl+h neighboring_window left
map ctrl+l neighboring_window right

# Unset the mapping to pass the keys to neovim
map --when-focus-on var:IS_NVIM ctrl+j
map --when-focus-on var:IS_NVIM ctrl+k
map --when-focus-on var:IS_NVIM ctrl+h
map --when-focus-on var:IS_NVIM ctrl+l

# the 3 here is the resize amount, adjust as needed
map alt+j kitten relative_resize.py down  3
map alt+k kitten relative_resize.py up    3
map alt+h kitten relative_resize.py left  3
map alt+l kitten relative_resize.py right 3

map --when-focus-on var:IS_NVIM alt+j
map --when-focus-on var:IS_NVIM alt+k
map --when-focus-on var:IS_NVIM alt+h
map --when-focus-on var:IS_NVIM alt+l

allow_remote_control yes
listen_on unix:/tmp/mykitty
```

This configuration allows me to switch between kitty windows using ctrl+hjkl, but I can't switch between nvim windows. The keystrokes seem to be intercepted by kitty - the plugin isn't activating unless I explicitly run a command to launch it.

When the plugin is active, I can navigate only between nvim windows. Meaning if I switch from a kitty window to a nvim window, I can't switch back.

What am I doing wrong?

**Comment by mrjones2014** [comment]
Is there anything relevant in the logs? `:SmartSplitsLog`

What versions of Neovim, the plugin, and Kitty are you using? I believe Kitty user variables are a somewhat newish feature so make sure you're using an up-to-date version of Kitty.

**Reply by klnstprx** [reply]
:SmartSplitsLog 
```
[Thu Sep  5 14:55:05 2024][smart-splits.nvim] Auto-detected multiplexer back-end: kitty
[Thu Sep  5 16:27:41 2024][smart-splits.nvim] Auto-detected multiplexer back-end: none
[Thu Sep  5 16:28:37 2024][smart-splits.nvim] Auto-detected multiplexer back-end: none
[Thu Sep  5 16:31:39 2024][smart-splits.nvim] Auto-detected multiplexer back-end: kitty
[Thu Sep  5 16:35:57 2024][smart-splits.nvim] Auto-detected multiplexer back-end: kitty
[Thu Sep  5 18:57:34 2024][smart-splits.nvim] Auto-detected multiplexer back-end: kitty
```
Nothing relevant I think.

❯ nvim -v
```
NVIM v0.10.1
Build type: Release
LuaJIT 2.1.1725453128
Run "nvim -V1 -v" for more info
```
❯ kitty -v
```
kitty 0.36.1 created by Kovid Goyal
```

And from what I understand from lazy.nvim I'm on the newest commit.
```
    ○ smart-splits.nvim  <C-l>  <C-j>  <C-h>  <C-k> 
        dir    /Users/klnstpr/.local/share/nvim/lazy/smart-splits.nvim
        url    https://github.com/mrjones2014/smart-splits.nvim
        branch master
        commit aee3093
        readme README.md
        help   |smart-splits.txt|
        keys    <C-l>  <C-j>  <C-h>  <C-k> 

```

**Reply by mrjones2014** [reply]
```
[Thu Sep  5 16:28:37 2024][smart-splits.nvim] Auto-detected multiplexer back-end: none
```

This would make it not work in Neovim, because it doesn't know which mux backend to use, but based on the logs, looks like that issue is resolved.

```
kitty 0.36.1 created by Kovid Goyal
```

This is the latest stable version, so looks like that's not the issue either...

And yeah, looks like you're on latest plugin version as well.

Not sure what's going on here. I don't have kitty installed at the moment (a lot of the kitty stuff is from community contributors), so I'll have to set aside some time to see if I can reproduce.

**Reply by klnstprx** [reply]
> ```
> [Thu Sep  5 16:28:37 2024][smart-splits.nvim] Auto-detected multiplexer back-end: none
> ```
> 
> This would make it not work in Neovim, because it doesn't know which mux backend to use, but based on the logs, looks like that issue is resolved.

Yeah I fiddled around a lot trying the most random stuff, but to no avail.

I will keep on trying, thanks.

**Reply by mrjones2014** [reply]
Out of curiosity, do any of the older versions work for you? https://github.com/mrjones2014/smart-splits.nvim/releases

**Reply by klnstprx** [reply, marked as answer]
Hi again, I have some updates. 
I came back home to my main linux desktop and started with a fresh install of nvim and lazyvim + smart-splits. I added the lines from README to my kitty config and installed smart-splits using the same snippet. This resulted in me being able to swap between kitty windows - couldn't swap windows inside nvim. I figured out that lazy.nvim wasn't loading the plugin on keystrokes. Setting "lazy" to false inside the plugin configuration got rid of the issue.

Forcing the plugin to load on startup also fixed the issue on my macOS system, where I initially encountered the problem.

Thanks for the help anyways!

new configuration:
```
return {
  {
    "mrjones2014/smart-splits.nvim",
    lazy = false,
    build = "./kitty/install-kittens.bash",
    keys = {
      {
        "<C-h>",
        function()
          require("smart-splits").move_cursor_left()
        end,
        desc = "Move to left window",
      },
      {
        "<C-l>",
        function()
          require("smart-splits").move_cursor_right()
        end,
        desc = "Move to right window",
      },
      {
        "<C-j>",
        function()
          require("smart-splits").move_cursor_down()
        end,
        desc = "Move to below window",
      },
      {
        "<C-k>",
        function()
          require("smart-splits").move_cursor_up()
        end,
        desc = "Move to above window",
      },
    },
  },
}
```

**Reply by mrjones2014** [reply]
Ah, right, I have this note under Wezterm: https://github.com/mrjones2014/smart-splits.nvim#wezterm

Kitty integration also somewhat recently  moved to using user variables, and so basically the issue is the Kitty config is looking for user variables, but if the plugin isnt loaded, the user variable isn't set. 

I'll make a note to add this to the README for Kitty as well.
